diff --git a/field_collection_table.module b/field_collection_table.module
index 26e64e8..4be7300 100644
--- a/field_collection_table.module
+++ b/field_collection_table.module
@@ -306,3 +306,27 @@ function field_collection_table_field_widget_form(&$form, &$form_state, $field,
   }
   return $element;
 }
+
+/**
+ * Implements hook_field_attach_form().
+ *
+ * Corrects #max_delta when we hide the blank field collection item.
+ *
+ * @see field_add_more_js()
+ * @see field_collection_field_widget_form()
+ */
+function field_collection_table_field_attach_form($entity_type, $entity, &$form, &$form_state, $langcode) {
+
+  foreach (field_info_instances($entity_type, $form['#bundle']) as $field_name => $instance) {
+    $field = field_info_field($field_name);
+
+    if ($field['type'] == 'field_collection' && $field['settings']['hide_blank_items']
+        && field_access('edit', $field, $entity_type) && $instance['widget']['type'] == 'field_collection_table') {
+
+      $element_langcode = $form[$field_name]['#language'];
+      if ($form[$field_name][$element_langcode]['#max_delta'] > 0) {
+        $form[$field_name][$element_langcode]['#max_delta']--;
+      }
+    }
+  }
+}
\ No newline at end of file
